presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java

```diff
@@ -572,7 +572,14 @@ public Block getBlock(int[] positions, int positionCount)
             nextPosition = positions[positionIndex];
         }
 
-        return ArrayBlock.fromElementBlock(positionCount, Optional.ofNullable(nullsCopy), offsetsCopy, elementStreamReader.getBlock(elementOutputPositions, elementOutputPositionCount));
+        Block elementBlock;
+        if (elementOutputPositionCount == 0) {
+            elementBlock = outputType.getElementType().createBlockBuilder(null, 0).build();
```

Comment:
Could you make a helper `private static Block createEmptyBlock(Type type)` and consolidate all other use cases in this file as well?